src/scripts/main.js

```diff
@@ -1,3 +1,194 @@
 'use strict';
 
-// write code here
+const tableHeaders = document.querySelectorAll('th');
+const rowsArray = Array.from(document.querySelectorAll('tbody tr'));
+const sortDirection = {};
+
+tableHeaders.forEach((header, index) => {
+  sortDirection[index] = true;
+
+  header.addEventListener('click', (e) => {
+    const sortedRows = rowsArray.sort((rowA, rowB) => {
+      const cellA = rowA.cells[index].textContent.trim();
+      const cellB = rowB.cells[index].textContent.trim();
+
+      if (index === 4) {
+        const salaryA = parseFloat(cellA.replace(/[$,]/g, ''));
+        const salaryB = parseFloat(cellB.replace(/[$,]/g, ''));
+
+        return sortDirection[index] ? salaryA - salaryB : salaryB - salaryA;
+      } else {
+        return sortDirection[index]
+          ? cellA.localeCompare(cellB)
+          : cellB.localeCompare(cellA);
+      }
```

Comment:
This sorting logic can be simplified. You can create a separate function to handle the sorting logic. This function can take the index and sortDirection as parameters and return a function that can be used in the sort method. This will make your code cleaner and easier to understand.